src/main/java/org/truffleruby/language/globals/GlobalVariables.java

```diff
@@ -28,7 +28,12 @@ public GlobalVariables(DynamicObject defaultValue) {
 
     @TruffleBoundary
     public GlobalVariableStorage getStorage(String name) {
-        return variables.computeIfAbsent(name, k -> new GlobalVariableStorage(defaultValue, null, null));
+        GlobalVariableStorage storage = variables.get(name);
+        if (storage == null) {
+            variables.putIfAbsent(name, new GlobalVariableStorage(defaultValue, null, null));
+            storage = variables.get(name);
+        }
+        return storage;
```

Comment:
I measured the overhead of having the lambda in the interpreter (since this is beyond a TruffleBoundary), see https://gist.github.com/eregon/303151e5a50b7b4805108f3fa42e2529.
The overhead is not negligible on HotSpot, but it is on GraalVM.